src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +36,35 @@
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${id}`}
             </div>
 
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => setSelectedTodo(null)}
+            />
           </header>
 
           <div className="modal-card-body">
             <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
+              {title}
             </p>
 
             <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
+              <strong
+                className={cn({
+                  'has-text-danger': !completed,
+                  'has-text-success': completed,
+                })}
+              >
+                {completed ? `Done` : 'Planned'}
+              </strong>
 
               {' by '}
 
-              <a href="mailto:Sincere@april.biz">Leanne Graham</a>
+              <a href={`mailto:${user?.email}`}>{user?.name}</a>
```

Comment:
You are interacting with the DOM directly by using the `href` attribute. In React, it's recommended to use the `Link` component from `react-router-dom`. This ensures that your app is a single-page application and that the browser does not make a full page refresh when the user navigates to a different page.